app/models/workflow_run.rb

```diff
@@ -198,6 +204,20 @@ def notification_params
     )
   end
 
+  def trigger_failed_reason
+    last_error = passports.where(reason: :trigger_failed, kind: :error).last
+    return if last_error.nil?
+
+    case last_error.metadata["error_reason"]
+    when "workflow_parameter_not_provided"
+      "Missing workflow parameter: #{last_error.metadata["error_message"]}"
+    when "workflow_dispatch_missing"
+      "Wrongly configured workflow: #{last_error.metadata["error_message"]}"
+    else
+      last_error.metadata["error_message"]
+    end
+  end
```

Comment:
I know there's been back-and-forth on this (and I think my suggestion on Slack was perhaps lost), but I wouldn't bother with reconstructing each message. I think the message in passport is sufficient:

> The %{kind} workflow run could not be triggered for ... Error – [actual reason from source]

This additional decoration doesn't add much value imo. It almost says the same thing differently. We do the same thing for [submission errors](https://github.com/tramlinehq/tramline/blob/4b663a8fedd6d75325b9a9cb9b965059d121e15c/app/components/live_release/pre_prod_release/submission_component.html.erb#L47).